src/main/java/de/medizininformatikinitiative/flare/model/fhir/Bundle.java

```diff
@@ -4,12 +4,13 @@
 
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public record Bundle(List<Entry> entry, List<Link> link) {
 
     public Bundle {
-        entry = entry == null ? List.of() : List.copyOf(entry);
+        entry = entry == null ? List.of() : List.copyOf(entry).stream().filter(Entry::hasReference).toList();
```

Comment:
See below: As we no longer have `hasReference` we should leave the entries here as they were before and use `bundle.entry().stream().flatMap(e -> e.resource().patientId().stream())` instead in `DataStore` line 56.